driver-core/src/main/java/com/datastax/driver/core/policies/RollingCount.java

```diff
@@ -0,0 +1,121 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core.policies;
+
+import io.netty.util.internal.chmv8.LongAdderV8;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLongArray;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * A "rolling" count over a 1-minute sliding window.
+ */
+class RollingCount {
+    // Divide the minute into 5-second intervals
+    private static final long INTERVAL_SIZE = TimeUnit.SECONDS.toNanos(5);
+
+    // A circular buffer containing the counts over the 12 previous intervals. Their sum is the count we're looking for.
+    // If we're at t = 61s this would span [0,60]
+    private final AtomicLongArray previousIntervals = new AtomicLongArray(12);
+    // The interval we're currently recording events for. It hasn't completed yet, so it's not included in the count.
+    // If we're at t = 61s this would span [60,65]
+    private final LongAdderV8 currentInterval = new LongAdderV8();
+    // Other mutable state, grouped in an object for atomic updates
+    private final AtomicReference<State> state;
+    private final Clock clock;
+
+    RollingCount(Clock clock) {
+        this.state = new AtomicReference<State>(new State(clock.nanoTime()));
+        this.clock = clock;
+    }
+
+    RollingCount() {
+        this(Clock.DEFAULT);
+    }
+
+    void increment() {
+        add(1);
+    }
+
+    void add(long amount) {
+        tickIfNecessary();
+        currentInterval.add(amount);
+    }
+
+    long count() {
+        tickIfNecessary();
+        return state.get().totalCount;
+    }
+
+    private void tickIfNecessary() {
+        State oldState = state.get();
+        long newTick = clock.nanoTime();
+        long age = newTick - oldState.lastTick;
+        if (age > INTERVAL_SIZE) {
```

Comment:
Time slots in the long array seem to be lower bound inclusive and upper bound exclusive, so shouldn't this comparison be `age >= INTERVAL_SIZE`?